src/main/java/seedu/intrack/logic/commands/DeleteCommand.java

```diff
@@ -31,6 +31,7 @@ public class DeleteCommand extends Command {
      * @param targetIndex Index of the internship application to be deleted.
      */
     public DeleteCommand(Index targetIndex) {
+        assert targetIndex.getOneBased() > 0 : "index should be a positive unsigned integer";
```

Comment:
While ultimately it doesn't matter as this is largely defensive, it might make more sense to put this after the `requireNonNull(targetIndex)` line to ensure the targetIndex is not null first.